src/cosi_db/model/household.rs

```diff
@@ -0,0 +1,73 @@
+use async_trait::async_trait;
+use mongodb::bson::oid::ObjectId;
+use mongodb::options::AggregateOptions;
+use mongodb::{bson::doc, bson::from_document, bson::Document, options::FindOptions};
+use rocket::futures::TryStreamExt;
+
+use names::Name;
+use rand::rngs::ThreadRng;
+use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+
+// cosi_db
+use crate::cosi_db::connection::CosiDB;
+use crate::cosi_db::controller::common::get_connection;
+
+use crate::cosi_db::model::address::Address;
+use crate::cosi_db::model::common::{COSICollection, Generator};
+use crate::cosi_db::model::person::Person;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct Household {
```

Comment:
Current implementation does not account for references. Require a modification to COSICollection to account for this.

Reply:
Updated `COSICollection` to support fetch and obtaining proper values.